homework/Task 2/Training.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace homework._Task2
+{
+    public class Training
+    {
+        private string _description;
+        private Lesson[] _resources = new Lesson[0];
+
+        public Training(string description)
+        {
+            _description = description;
+        }
+
+        public void Add(Lesson lesson)
+        {
+            int resourcesSize = _resources.Length + 1;
+            Lesson[] tempResources = new Lesson[resourcesSize];
+            for (int index = 0; index < _resources.Length; index++)
+            {
+                tempResources[index] = _resources[index];
+            }
+            tempResources[resourcesSize - 1] = lesson;
+            _resources = tempResources;
+        }
+
+        public bool IsPractical()
+        {
+            foreach (var resource in _resources) 
+            {
+                if (!(resource is PracticalLesson))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public Training Clone()
+        {
+            Training clone = (Training) this.MemberwiseClone();
+            clone._description = string.Copy(_description);
```

Comment:
Sorry, but this is not a deep cloning, see my lectures. In this case you call Clone() from array, and this method copy only references to Lessons, not the Lesson objects.